### ISSUE TITLE
Iterating hexes in a ring effect is implemented in correctly

### ISSUE THREAD

**Issue by arlith** [MegaMek/megamek#440]
In some cases, we use code like the following:

```
for (int dist = 1; dist < maxDist; dist++) {
    for (int dir = 0; dir < 6; dir++) {
        Coords c = position.translated(dir, dist);
        ....
    }
}
```

This looks correct, but it's not.  It works for "adjacent" hexes (dist = 1), but it doesn't work further out than that.  At the first "ring" there will be 6 hexes, but at the second "ring" there will be 12 hexes,  24, and so on.  Ultimately, this leads to the "start" pattern witnessed in #434:

![image](https://cloud.githubusercontent.com/assets/19868605/21044109/437a004c-bdc8-11e6-8931-3bee1cdb49d2.png)

There are at least three spots I'm immediately aware of where this is an issue:

Server.doExplosion, at line 23060
Server.getAPDSProtectedCoords, at line 13320
Compute.coordsAtRange, at line 5072

It's possible that there are more cases of this as well.

**Comment by VampireSeraphin** [comment]
Will this also effect very large artillery explosions like cruise missiles. Are Command line nukes effected?

**Comment by arlith** [comment]
More than likely, yes.

**Comment by arlith** [comment]
I should've posted this a while ago, but this website describes the algorithm we should be using: http://www.redblobgames.com/grids/hexagons/.  Our current code is based off of the information in this website as well, but some of the names were changed.

**Comment by arlith** [comment]
I think this is far less systemic than I had initially thought.  Turns out `Compute.coordsAtRange` is correct, and does what it describes (aka, provides a ring of `Coord`s at the supplied range from the supplied center hex).  The problem, as far I could find was only in `Server.doExplosion` and `Server.getAPDSProtectedCoords` (likely because I used the code in `Server.doExplosion` as a template when writing the APDS code).

I searched the rest of the code for `for (int dir =` and looked at any calls to `Coord.translate` after that kind of loop.  I didn't find any instances where the translate call was using a distance (this pattern works for adjacent hexes).  So, I think that I have caught all cases of improper usage.
